### ISSUE TITLE
Copying text to the host from (pure) Wayland clients doesn't work on Fedora 32

### ISSUE THREAD

**Issue by eleses** [vmware/open-vm-tools#443]
Since Fedora 31, [Qt Wayland is on by default](https://fedoraproject.org/wiki/Changes/Qt_Wayland_By_Default_On_Gnome), meaning that Qt apps on Fedora 31+ don't use `xcb` anymore by default. Alas open-vm-tools seems to only work on xcb clients when it come to copying text of them. 

On Fedora 32, if I start a Qt app (e.g. `qjackctl`) with the default settings, then I cannot copy out text of it to the VMware host (copying inside the VM from one app to another works fine still), but if I start the app with `QT_QPA_PLATFORM=xcb` which forces Xwayland fallback in Qt, then it does work to copy out from the (Qt X11) app to the VM host.

I also see that

```
$ xlsclients 
localhost.localdomain  ibus-x11
localhost.localdomain  vmware-user
localhost.localdomain  gsd-xsettings
localhost.localdomain  gnome-shell
```

which suggest that the vmware tools only run under Xwayland.

**Comment by dsouders** [comment]
Yes, this is a known issue.  Wayland support is currently under development.

**Comment by cmprmsd** [comment]
Is there any status update on this issue? :)

**Comment by BenjaminQuandt** [comment]
This Issue is older than 2 yaears now! any news?

**Comment by Slapbox** [comment]
Text works on my Kubuntu machine with Wayland, but images do not.

**Comment by BenjaminQuandt** [comment]
I run it with my Manjaro. For now i found a workaround to be able to work. sadly without wayland. Manjaro has the option to switch on x11 on the bootscreen. Here I still have problems. 
But if i boot in my Manjaro with x11 and execute vmware-user i can use full c&p mounts and so on till i restart. The downside is i have to run the vmware-user on every restart and i cant put it in ausostart. Also i had to enable a service that was disabled by default. It looks like there are stange gtk calls in the background that let u run the stuff.

Anyways i still wait fur a functional wayland integration. But i don't have high hopes for that. I guess a lot of stuff has to be written new for that. Some Informations from the devs would be helpfull here. even if it is a "it will take at least..." or a "we will not support it".

**Comment by cmprmsd** [comment]
I use copyQ which kind of works. You have to move the mouse to the GUI of VMware first and then move it to Wayland native windows. That works on Sway at least. If you move the mouse to fast, X has no time to sync the clipboard, which causes the sync to the Wayland clipboard to fail. 

VMware should just allow for a Wayland native GUI. That would work wonders.

**Comment by Slapbox** [comment]
> The downside is i have to run the vmware-user on every restart and i cant put it in ausostart. 

Same here with Kubuntu 22.04 and above. See #568.

**Comment by heheda123123** [comment]
2023-5 archlinux still have this problem

**Comment by nzlov** [comment]
now. 2024-11 archlinux still have this problem

**Comment by heheda123123** [comment]
virtualbox has now fixed this problem and supports copying in wayland

**Comment by cmprmsd** [comment]
Sadly commercial licenses for Virtualbox extensions come in 100+ packs. Thank you Oracle for not providing single seat licenses. 😢
